site-builder/src/site/resource.rs

```diff
@@ -344,15 +344,18 @@ impl ResourceManager {
             extension.to_string_lossy(),
             full_path.to_string_lossy()
         ))?)?;
-        // TODO(giac): this could be (i) async; (ii) pre configured with the number of shards to
-        //     avoid chain interaction (maybe after adding `info` to the JSON commands).
-        let output = self.walrus.blob_id(full_path.to_owned(), None)?;
-        // Need to check the size to avoid calling encode on empty files.
+
         let plain_content = std::fs::read(full_path)?;
         if plain_content.is_empty() {
-            // We are ignoring empty files.
-            return Ok(None);
+            let error_message = format!(
+                "The file {} is empty. Walrus does not support empty files.",
```

Comment:
Nit: I just noticed that here and in other parts of walrus sites we use capitalized errors. However, it would be best to follow the stdlib convention (which I believe we use in walrus too) of having error messages as "concise lowercase sentences without trailing punctuation:" (see [docs](https://doc.rust-lang.org/stable/std/error/trait.Error.html))
```suggestion
                "the file {} is empty; Walrus does not support empty files",
```